hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/store/contracts/precompile/HRCPrecompilesTest.java

```diff
@@ -0,0 +1,585 @@
+        dissociation = Mockito.mockStatic(Dissociation.class);
+    }
+
+    @AfterEach
+    void closeMocks() {
+        entityIdUtils.close();
+        association.close();
+        dissociation.close();
+    }
+
+    @Test
+    @DisplayName("Associate token via HRC facade")
+    void hrcAssociate() {
+        final Bytes nestedPretendArguments = Bytes.of(Integers.toBytes(ABI_ID_HRC_ASSOCIATE));
+        final Bytes pretendArguments = givenFrameContext(nestedPretendArguments);
+
+        givenLedgers();
+        givenPricingUtilsContext();
+
+        given(dynamicProperties.isHRCAssociateEnabled()).willReturn(true);
+        given(wrappedLedgers.tokens().exists(any())).willReturn(true);
+        given(infrastructureFactory.newSideEffects()).willReturn(sideEffects);
+        given(worldUpdater.permissivelyUnaliased(any()))
+                .willAnswer(invocationOnMock -> invocationOnMock.getArgument(0));
+        given(Association.singleAssociation(any(), any())).willReturn(HTSTestsUtil.associateOp);
+        given(syntheticTxnFactory.createAssociate(HTSTestsUtil.associateOp)).willReturn(mockSynthBodyBuilder);
+        given(infrastructureFactory.newAccountStore(accounts)).willReturn(accountStore);
+        given(infrastructureFactory.newTokenStore(accountStore, sideEffects, tokens, nfts, tokenRels))
+                .willReturn(tokenStore);
+        given(infrastructureFactory.newAssociateLogic(accountStore, tokenStore)).willReturn(associateLogic);
+        given(associateLogic.validateSyntax(any())).willReturn(OK);
+        given(sigsVerifier.hasActiveKey(
+                        false,
+                        Id.fromGrpcAccount(accountMerkleId).asEvmAddress(),
+                        HTSTestsUtil.senderAddress,
+                        wrappedLedgers,
+                        TokenAssociateToAccount))
+                .willReturn(true);
+        given(feeCalculator.estimatedGasPriceInTinybars(HederaFunctionality.ContractCall, HTSTestsUtil.timestamp))
+                .willReturn(1L);
+        given(mockSynthBodyBuilder.build())
+                .willReturn(TransactionBody.newBuilder().build());
+        given(mockSynthBodyBuilder.setTransactionID(any(TransactionID.class))).willReturn(mockSynthBodyBuilder);
+        given(feeCalculator.computeFee(any(), any(), any(), any())).willReturn(mockFeeObject);
+        given(mockFeeObject.serviceFee()).willReturn(1L);
+        given(worldUpdater.aliases()).willReturn(aliases);
+        given(aliases.resolveForEvm(any())).willAnswer(invocationOnMock -> invocationOnMock.getArgument(0));
+        given(creator.createSuccessfulSyntheticRecord(Collections.emptyList(), sideEffects, EMPTY_MEMO))
+                .willReturn(mockRecordBuilder);
+
+        // when:
+        subject.prepareFields(frame);
+        subject.prepareComputation(pretendArguments, a -> a);
+        subject.getPrecompile().getGasRequirement(HTSTestsUtil.TEST_CONSENSUS_TIME);
```

Comment:
Does this line _do_ anything?  (Calls two getters in a row ... or at least `getGasRequirement` _looks_ like a getter)

Reply:
No - doesn't do anything.  I took it out and also removed dependent mocks here and below.